### PR TITLE
Converted aggregate_metric_double field mapper to parameterized mapper

### PR DESCRIPTION
**Pull request by csoulios**
`AggregateDoubleMetricFieldMapper` was originally not written as a parametrized field mapper.

This PR refactors `AggregateDoubleMetricFieldMapper` to extend `ParametrizedFieldMapper`.

Also, refactored `AggregateDoubleMetricFieldMapperTests` to extend `MapperTestCase` instead of `ESSingleNodeTestCase`

### CODE REVIEW

File: x-pack/plugin/mapper-aggregate-metric/src/main/java/org/elasticsearch/xpack/aggregatemetric/mapper/AggregateDoubleMetricFieldMapper.java [hunk 1]
```diff
@@ -102,95 +101,89 @@ public static String subfieldName(String fieldName, Metric metric) {
...
-                throw new IllegalArgumentException(
-                    "Property [" + Names.METRICS.getPreferredName() + "] must be set for field [" + name() + "]."
+            if (defaultMetric.isConfigured() == false) {
+                // If a single metric is contained, this should be the default
+                if (metrics.getValue().size() == 1) {
+                    Metric m = metrics.getValue().iterator().next();
+                    defaultMetric.setValue(m);
+                }
+
+                if (metrics.getValue().contains(defaultMetric.getValue()) == false) throw new IllegalArgumentException(
```
Comment (romseygeek) [inline x-pack/plugin/mapper-aggregate-metric/src/main/java/org/elasticsearch/xpack/aggregatemetric/mapper/AggregateDoubleMetricFieldMapper.java:174]: nit: can we have `{}` around the `throw` clause?

**Review by romseygeek — APPROVED** [review]
LGTM, thanks @csoulios

**Comment by csoulios** [comment]
@elasticmachine run elasticsearch-ci/2
@elasticmachine run elasticsearch-ci/bwc

**Comment by csoulios** [comment]
@elasticmachine run elasticsearch-ci/bwc

**Comment by csoulios** [comment]
@elasticmachine run elasticsearch-ci/2

**Pull request merged**